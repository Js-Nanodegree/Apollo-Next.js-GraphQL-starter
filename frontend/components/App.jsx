import React, { Component } from 'react';

import Head from 'next/head';
import Loading from './Loading';
import Navigation from './Navigation';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import withData from '../lib/withData';
import { withRouter } from 'next/router';

const App = ({ children, title, link, meta, showNavigation, Me }) => (
  <div className="fullHeight">
    <Head>
      <title>{title}</title>
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      <link
        href="https://fonts.googleapis.com/css?family=Roboto:300,400"
        rel="stylesheet"
      />
    </Head>

    {showNavigation ? <Navigation title={title} Me={Me} /> : null}

    <main id="main">{children}</main>
  </div>
);

export default App;
