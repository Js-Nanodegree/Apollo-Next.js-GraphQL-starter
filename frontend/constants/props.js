import { shape, string } from 'prop-types';

export const user = shape({
  __typename: string.isRequired,
  _id: string.isRequired,
  email: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired
});
