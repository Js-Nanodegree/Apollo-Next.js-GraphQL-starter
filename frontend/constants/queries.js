import gql from 'graphql-tag';

export const CURRENT_USER_QUERY = gql`
  query {
    Me {
      _id
      email
      firstName
      lastName
    }
  }
`;
