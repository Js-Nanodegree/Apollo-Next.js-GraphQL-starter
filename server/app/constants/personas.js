export default [
    {level: 0, label: "Assistant"},
    {level: 0, label: "Junior Manager"},
    {level: 0, label: "Manager"},
    {level: 0, label: "Analyst"},
    {level: 0, label: "Category Manager"},
    {level: 0, label: "Category Director"},
    {level: 0, label: "Senior Manager"},
    {level: 0, label: "Performance"},
    {level: 0, label: "Director"},
    {level: 0, label: "Head of"},
    {level: 0, label: "Head of Excellence"},
    {level: 0, label: "Head of Indirect"},
    {level: 0, label: "Head of Direct"},
    {level: 0, label: "VP"},
    {level: 0, label: "SVP"},
    {level: 0, label: "CPO"},
    {level: 0, label: "Global CPO"},
    {level: 0, label: "C-Level"},
    {level: 0, label: "Consultant"},
    {level: 0, label: "Partner"},
    {level: 0, label: "Lecturer"},
    {level: 0, label: "Student"},
    {level: 0, label: "Journalist"},
    {level: 0, label: "Buyer"}
]