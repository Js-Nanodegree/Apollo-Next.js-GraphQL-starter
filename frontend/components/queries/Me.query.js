import { shape, string, func } from 'prop-types';
import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from '../../constants/queries';

export const MeProps = shape({
  __typename: string.isRequired,
  _id: string.isRequired,
  email: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired
});

const Me = props => (
  <Query {...props} query={CURRENT_USER_QUERY}>
    {payload => props.children(payload)}
  </Query>
);

Me.propTypes = {
  children: func.isRequired
};

export default Me;
