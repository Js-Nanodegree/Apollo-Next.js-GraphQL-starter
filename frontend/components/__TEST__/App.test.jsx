import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { CURRENT_USER_QUERY } from '../../constants/queries';
import App from '../App';

/*
 * Next router can only be used on the client so it needs to be mocked
 */
jest.mock('next/router');

const mocks = [
  {
    request: {
      query: CURRENT_USER_QUERY
    },
    result: {
      __typename: 'User',
      _id: '123456',
      email: 'test@example.com',
      firstName: 'Test',
      lastName: 'Example'
    }
  }
];

describe('App component', () => {
  it('should match snap shot', () => {
    const title = 'Test title';

    const withNavigation = renderer
      .create(
        <MockedProvider mocks={mocks}>
          <App title={title} showNavigation>
            <div>
              <p>Test app children</p>
            </div>
          </App>
        </MockedProvider>
      )
      .toJSON();

    expect(withNavigation).toMatchSnapshot();

    const noNavigation = renderer
      .create(
        <App title={title} showNavigation={false}>
          <div>
            <p>Test app children</p>
          </div>
        </App>
      )
      .toJSON();

    expect(noNavigation).toMatchSnapshot();
  });

  it('should render the page title', () => {
    const title = 'Test title';

    const wrapper = shallow(
      <App title={title} showNavigation={false}>
        <div>
          <p>Test app children</p>
        </div>
      </App>
    );

    expect(wrapper.find('title').text()).toBe(title);
  });
});
