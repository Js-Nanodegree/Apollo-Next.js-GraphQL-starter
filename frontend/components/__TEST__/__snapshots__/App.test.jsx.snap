// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snap shot 1`] = `
<div
  className="fullHeight"
>
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 nav-container MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8 ff7z3s-0 ixKqbe"
  >
    <div
      className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
      <p
        className="MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorInherit-70"
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        Test title
      </p>
      <button
        className="MuiButtonBase-root-102 MuiButton-root-76 MuiButton-text-78 MuiButton-flat-81"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-77"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-105"
        />
      </button>
      <button
        className="MuiButtonBase-root-102 MuiButton-root-76 MuiButton-text-78 MuiButton-flat-81"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-77"
        >
          Register
        </span>
        <span
          className="MuiTouchRipple-root-105"
        />
      </button>
    </div>
  </header>
  <main
    id="main"
  >
    <div>
      <p>
        Test app children
      </p>
    </div>
  </main>
</div>
`;

exports[`App component should match snap shot 2`] = `
<div
  className="fullHeight"
>
  <main
    id="main"
  >
    <div>
      <p>
        Test app children
      </p>
    </div>
  </main>
</div>
`;
